lib/arel_extensions/visitors/mssql.rb

```diff
@@ -155,10 +155,19 @@ def visit_ArelExtensions_Nodes_Duration o, collector
       end
 
       def visit_ArelExtensions_Nodes_Length o, collector
-        collector << "#{o.bytewise ? 'DATALENGTH' : 'LEN'}("
-        collector = visit o.expr, collector
-        collector << ")"
-        collector
+        if o.bytewise
```

Comment:
Could you please add some comments explaining what's going on here?  Maybe a url to the relevant documentation.

Reply:
And it seems to be duplicating its argument.  Is this guaranteed to be harmless?